client/components/ReviewsOverview.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ReviewsOverview = {
+
+  reviewScores: [5, 4, 3, 2, 1],
+
+  totalReviewsAverageRating(reviews) {
+    return (reviews.reduce((acc, review) => acc + (review.rating || 0), 0)
+    / reviews.length).toFixed(1);
```

Comment:
Hey, when averaging the ratings, you appear to be adding a 0 for reviews that don't have ratings but then dividing by the total number of reviews.
i.e.
Line 9: acc + (review.rating || 0 )
Line 10: / reviews.length )
Correct me if I'm reading that wrong.

If a large number of reviews don't give ratings, this could significantly drag down your averages. Wouldn't it be more accurate to only divide by the number of reviews that actually have ratings? You could implement some sort of counter in the reduce function and then divide by that counter.

Then again, if you're not anticipating a large number of reviews that leave off the rating (or if it's not possible to leave a review without a rating) this is a moot point.

Reply:
Great find! You have to leave a rating score, but I will refactor this to remove the `||` so that the code is somewhat future proof!